manifests/mod/ssl.pp

```diff
@@ -52,11 +53,6 @@
     }
   }
 
-  validate_bool($ssl_compression)
-
-  validate_array($ssl_proxy_protocol)
-  validate_string($ssl_sessioncache)
-
   if is_bool($ssl_honorcipherorder) {
```

Comment:
Maybe this can be one big switch statement that includes ```true``` and ```false``` because I believe ```is_bool``` also emits a deprecation warning.